ui/raidboss/emulator/EventBus.ts

```diff
@@ -0,0 +1,65 @@
+type CallbackFunction = () => void;
+type EventMapEntry = {
+  event: string;
+  // Disable ban-types due to scope being any object type
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  scope: object;
+  callback: CallbackFunction;
+};
+type EventMap = {[event: string]: EventMapEntry[]};
+
+/**
+ * This is a base class that classes can extend to inherit event bus capabilities.
+ * This allows other classes to listen for events with the `on` function.
+ * The inheriting class can fire those events with the `dispatch` function.
+ */
+export default class EventBus {
+  private listeners: EventMap = {};
+  /**
+   * Subscribe to an event
+   *
+   * @param event The event(s) to subscribe to, space separated
+   * @param callback The callback to invoke
+   * @param scope Optional. The scope to apply the function against
+   * @returns The callbacks registered to the event(s)
+   */
+  on(event: string,
+      callback: CallbackFunction | undefined = undefined,
+      // Disable ban-types due to scope being any object type
+      // eslint-disable-next-line @typescript-eslint/ban-types
+      scope: object | undefined = undefined,
```

Comment:
```suggestion
      callback?: CallbackFunction,
      scope?: Scope,
```

Along with the above suggestion for `Scope`, and I think `| undefined = undefined` can just be `?`.